oc-chef-pedant/lib/pedant/rspec/cookbook_util.rb

```diff
@@ -330,7 +330,11 @@ def verify_checksum_url(url, expected_reponse_code)
           http.ca_file     = Pedant::Config.ssl_ca_file     if Pedant::Config.ssl_ca_file
         end
 
-        response = http.get(uri.request_uri, {})
+        # NOTE(ssd) 2020-06-15: Chef::ServerAPI always sets the Host
+        # header to HOSTNAME:PORT. We do the same here to avoid v4
+        # signing issues.
```

Comment:
How far back does that behavior go? I think it's been in there a while, but it would be worth noting that if we being stricter than we used to. Pedant has been used as a guide for third party API libraries, so they may have incorporated the behavior. This is *not* intended to block this PR, but more of a thing we will have to note when the V4 signing work merges, and probably call out in the release notes, as it might break things (even if they are unsupported, its always nice to note when the contract changes).

Reply:
Yup, the V4 signing is definitely going to break some clients unfortunately.  When I looked at the issue last, I don't see a way around it for S3-backed chef-server.  For bookshelf-backed chef-servers we have some flexibility because we can do the auth check using both headers and accept either.

Reply:
Note that I _think_ all supported versions of Chef client should be OK but we should double check that, understand how far back users can be while still working, and document the issue for 3rd party clients.

Reply:
Note, when I say "break some clients" I mean clients broadly.  I think all supported versions of Chef will be OK, but we'll definitely want to document it for 3rd party clients and confirm which versions of Chef-Client work with v4 signing.

Reply:
Just to provide this FYI, I did 'hand testing' of the last 3-4 Chef clients a while back (manual knife cookbook uploads and downloads and what not).  Those all passed for the testing I did.